src/app/core/registry/registry.service.ts

```diff
@@ -220,4 +258,224 @@ export class RegistryService {
     );
   }
 
+  public editMetadataSchema(schema: MetadataSchema) {
+    this.store.dispatch(new MetadataRegistryEditSchemaAction(schema));
+  }
+
+  public cancelEditMetadataSchema() {
+    this.store.dispatch(new MetadataRegistryCancelSchemaAction());
+  }
+
+  public getActiveMetadataSchema(): Observable<MetadataSchema> {
+    return this.store.pipe(select(editMetadataSchemaSelector));
+  }
+
+  public selectMetadataSchema(schema: MetadataSchema) {
+    this.store.dispatch(new MetadataRegistrySelectSchemaAction(schema))
+  }
+
+  public deselectMetadataSchema(schema: MetadataSchema) {
+    this.store.dispatch(new MetadataRegistryDeselectSchemaAction(schema))
+  }
+
+  public deselectAllMetadataSchema() {
+    this.store.dispatch(new MetadataRegistryDeselectAllSchemaAction())
+  }
+
+  public getSelectedMetadataSchemas(): Observable<MetadataSchema[]> {
+    return this.store.pipe(select(selectedMetadataSchemasSelector));
+  }
+
+  public editMetadataField(field: MetadataField) {
+    this.store.dispatch(new MetadataRegistryEditFieldAction(field));
+  }
+
+  public cancelEditMetadataField() {
+    this.store.dispatch(new MetadataRegistryCancelFieldAction());
+  }
+
+  public getActiveMetadataField(): Observable<MetadataField> {
+    return this.store.pipe(select(editMetadataFieldSelector));
+  }
+
+  public selectMetadataField(field: MetadataField) {
+    this.store.dispatch(new MetadataRegistrySelectFieldAction(field))
+  }
+
+  public deselectMetadataField(field: MetadataField) {
+    this.store.dispatch(new MetadataRegistryDeselectFieldAction(field))
+  }
+
+  public deselectAllMetadataField() {
+    this.store.dispatch(new MetadataRegistryDeselectAllFieldAction())
+  }
+
+  public getSelectedMetadataFields(): Observable<MetadataField[]> {
+    return this.store.pipe(select(selectedMetadataFieldsSelector));
+  }
+
+  public createOrUpdateMetadataSchema(schema: MetadataSchema): Observable<MetadataSchema> {
```

Comment:
Large methods like this one need either some detailed Typedocs or (more) inline comments. (It's good to see a few small inline comments already, but I'd recommend adding a few more, or adding TypeDocs)